lib/chits.rb

```diff
@@ -0,0 +1,22 @@
+require 'pg'
+
+class Chits
+    def self.all
+        if ENV['ENVIRONMENT'] == 'test'
+            connection = PG.connect(dbname: 'Chitter_test')
+
+        else 
+            connection = PG.connect(dbname: 'Chitter')  
+           
+        end    
+     result = connection.exec('SELECT * FROM messages;')
+     
+
+    result.map{ |chitter|
+        puts chitter
+         chitter['messages']
+         }
```

Comment:
I can see you were part way through debugging this method. I found that this method doesn't return anything, by writing `puts chitter` in chitter_spec.rb.

Reply:
Further debugging here:

```
result.map{ |chitter|
   puts "chitter = " + chitter.to_s
   puts "chitter['messages'] = " + chitter['messages'].to_s
}
```
results in:
```
chitter = {"id"=>"61", "message"=>"Hey you! this is Chitter"}
chitter['messages'] =
chitter = {"id"=>"62", "message"=>"Have something to Chit?"}
chitter['messages'] =
chitter = {"id"=>"63", "message"=>"Chit it now!"}
chitter['messages'] =
```

Reply:
I think that's enough clues to start you off. If you'd like more, show me what you've tried.